lucille-core/src/main/java/com/kmwllc/lucille/connector/jdbc/DatabaseConnector.java

```diff
@@ -118,15 +118,17 @@ private int getIdColumnIndex(String[] columns) throws ConnectorException {
 
   @Override
   public void execute(Publisher publisher) throws ConnectorException {
-
+    ResultSet rs = null;
+    ArrayList<ResultSet> otherResults = null;
+    Connection connection = null;
     try {
-      // connect to the database.
-      Connection connection = createConnection();
+      connection = createConnection();

```

Comment:
where is the connection closed? Am I missing it? Can we declare the connection within the above try? i.e. 
```
try(Connection connection = createConnection) {
...
```

Reply:
Ahh okay I thought the connection was declared within the method but now I see that is removed code. Seems good then.

Reply:
I think Rudi commented already, but it's closed in the .close() method the number of connections created is not known until we collect them all into the globally defined array.